### PR TITLE
Fix #328. Merge plugin upstream master into 202

### LINKED ISSUE #269: [Rider][ReportedByUser] Azure Services Explorer - Error Refresh resource manage module
IntelliJ build version: 2019.2.3 RD-192.7317.11
OS: Mac OS X
JDK: JetBrains s.r.o 11.0.4
Plugin version: 3.26.0.565-2019.2
Error message: 
```
com.microsoft.azuretools.adauth.AuthException: No subscription found in the account
	at com.microsoft.azuretools.authmanage.SubscriptionManager.doSetSubscriptionDetails(SubscriptionManager.java:122)
	at com.microsoft.azuretools.authmanage.SubscriptionManager.updateSubscriptionDetailsIfNull(SubscriptionManager.java:92)
	at com.microsoft.azuretools.authmanage.SubscriptionManager.getSubscriptionDetails(SubscriptionManager.java:76)
	at com.microsoft.azuretools.utils.AzureModelController.updateSubscriptionMaps(AzureModelController.java:270)
	at com.microsoft.azuretools.core.mvp.model.AzureMvpModel.getResourceGroups(AzureMvpModel.java:130)
	at com.microsoft.tooling.msservices.serviceexplorer.azure.arm.ResourceManagementModulePresenter.onModuleRefresh(ResourceManagementModulePresenter.java:47)
	at com.microsoft.tooling.msservices.serviceexplorer.azure.arm.ResourceManagementModule.refreshItems(ResourceManagementModule.java:58)
	at com.microsoft.tooling.msservices.serviceexplorer.RefreshableNode.refreshItems(RefreshableNode.java:95)
	at com.microsoft.tooling.msservices.serviceexplorer.RefreshableNode$2.run(RefreshableNode.java:161)
	at com.microsoft.intellij.helpers.IDEHelperImpl$2$1.run(IDEHelperImpl.java:142)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgre

### LINKED ISSUE #281: [Rider][ReportedByUser] MS Azure Explorer - Error Loading Web Apps
IntelliJ build version: 2019.3.4 RD-193.6494.48
OS: Windows 10
JDK: JetBrains s.r.o 11.0.5
Plugin version: 3.26.0.607-2019.3
Error message: 
```
com.microsoft.azure.management.appservice.DefaultErrorResponseException: Status code 400, {"error":{"code":"DisallowedOperation","message":"The current subscription type is not permitted to perform operations on any provider namespace. Please use a different subscription."}}
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.microsoft.rest.ServiceResponseBuilder.build(ServiceResponseBuilder.java:122)
	at com.microsoft.azure.AzureResponseBuilder.build(AzureResponseBuilder.java:56)
	at com.microsoft.azure.management.appservice.implementation.WebAppsInner.listDelegate(WebAppsInner.java:1802)
	at com.microsoft.azure.management.appservice.implementation.WebAppsInner.access$000(WebAppsInner.java:59)
	at com.microsoft.azure.management.appservice.implementation.WebAppsInner$5.call(WebAppsInner.java:1789)
	at com.microsof

### LINKED ISSUE #282: [Rider][ReportedByUser] MS Azure Explorer - Error Loading Web Apps
IntelliJ build version: 2020.1 EAP 5 RD-201.6251.35
OS: Windows 10
JDK: JetBrains s.r.o 11.0.6
Plugin version: 3.26.0.621-2020.1
Error message: 
```
java.lang.RuntimeException: com.microsoft.azuretools.adauth.AuthException: Null refresh token in auth result
	at rx.exceptions.Exceptions.propagate(Exceptions.java:57)
	at rx.observables.BlockingObservable.subscribe(BlockingObservable.java:497)
	at com.microsoft.azure.management.resources.fluentcore.utils.PagedListConverter.loadConvertedList(PagedListConverter.java:100)
	at com.microsoft.azure.management.resources.fluentcore.utils.PagedListConverter.convert(PagedListConverter.java:66)
	at com.microsoft.azure.management.appservice.implementation.WebAppsImpl.wrapList(WebAppsImpl.java:100)
	at com.microsoft.azure.management.resources.fluentcore.arm.collection.implementation.TopLevelModifiableResourcesImpl.list(TopLevelModifiableResourcesImpl.java:118)
	at com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel.listWebApps(AzureWebAppMvpModel.java:565)
	at com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel.lambda$null$1(AzureWebAppMvpModel.java:524)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber.call(OperatorSubscribeOn.java:100)
	at rx.internal.schedulers.CachedThreadScheduler$EventLoopWorker$1.call(CachedThreadScheduler.java:230

### LINKED ISSUE #328: Merge upstream into Rider toolkit 202 branch
Upstream seems to have a couple of fixes we could benefit from.

#269, #281, #282 (fixed with microsoft/azure-tools-for-java#4479)

Also supports Azure CLI for credentials (microsoft/azure-tools-for-java#4468), which seems like it would magically fix #204 (needs verification).

### PR DESCRIPTION
**Pull request by sdubov**
* Fix Utils, Plugin modules
* Move AzureActionsListener into extension to support correct services registration for Java and Rider
* Fix plugin.xml - split actions to a separate groups, merged missing actions and clean up idea plugin.xml file (remove actions that were registered in general platformPlugin.xml file)
* Fix Node2Actions map for Java and Rider plugins
* Fix build.gradle - move :idea and general logic into plugin.xml (idea) and platformPlugin.xml accordingly, and move dependent plugin versions into gradle.properties
* Fix package naming and code location for Functions

Fixes #328
Fixes #269
Fixes #281
Fixes #282

### CODE REVIEW

**Comment by sdubov** [comment]
Couple notes:
1. Storage Account node is failing at runtime with getting a dependency from `azure-storage`:
```
java.lang.NoClassDefFoundError: Could not initialize class com.microsoft.azure.storage.core.Utility
```
2. Azure Streaming Logs tool window is added. It miss the icon - ([RIDER-48520](https://youtrack.jetbrains.com/issue/RIDER-48520)), but I believe we can include it in a release as well.

**Comment by maartenba** [comment]
LGTM apart from those two issues

**Review by maartenba — CHANGES_REQUESTED** [review]
Storage node expansion seems a blocker

File: .github/no-response.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+daysUntilClose: 14
```
Comment (maartenba) [inline .github/no-response.yml]: This will auto-close issues. Is this something we want to do?

**Comment by sdubov** [comment]
> Storage node expansion seems a blocker

+1

File: .github/no-response.yml (hunk 1 above)
Comment (sdubov) [reply .github/no-response.yml]: Definitely not. Thank you for the catch.

**Review by maartenba — APPROVED** [review]

**Pull request merged**